python312Packages.pytest-examples also fails to build on current master:

```
pytest_examples/traceback.py:81: DeprecationWarning
=========================== short test summary info ============================
FAILED tests/test_run_examples.py::test_run_example_ok_fail - AssertionError: assert ['test_run_ex...arnings.html'] == ['', '>   ass...se...
FAILED tests/test_run_examples.py::test_run_directly - DeprecationWarning: co_lnotab is deprecated, use co_lines instead.
============ 2 failed, 69 passed, 1 deselected, 1 xfailed in 2.76s =============
```

building python312Packages.pytest-examples on [1c37aef](https://github.com/NixOS/nixpkgs/commit/1c37aef70fa4c05c94e7ba916d3e09a8959a03c8) fails with the same test failures.

--- a/pkgs/development/tools/ruff/default.nix
+++ b/pkgs/development/tools/ruff/default.nix
@@ -10,16 +10,16 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "ruff";
-  version = "0.2.2";
+  version = "0.3.0";
 
   src = fetchFromGitHub {
     owner = "astral-sh";
     repo = "ruff";
     rev = "refs/tags/v${version}";
-    hash = "sha256-wCjPlKlw0IAh5oH4W7DUw3KBxR4bt9Ho7ncRL5TbD/0=";
+    hash = "sha256-U77Bwgbt2T8xkamrWOnOpNRF+8skLWhX8JqgPqowcQw=";
   };
 
-  cargoHash = "sha256-EHAlsEh3YnAhjIGC9rSgyK3gbKPCJqI6F3uAqZxv2nU=";
+  cargoHash = "sha256-IBcZRElbeu7Ab/7Q7N5TLhAznXxKsupifR83gfpY61Q=";
 
   nativeBuildInputs = [
     installShellFiles

